src/Notification/NotificationSyncer.php

```diff
@@ -94,6 +99,10 @@ public function sync(Blueprint\BlueprintInterface $blueprint, array $users)
             $this->setDeleted($toUndelete, false);
         }
 
+        foreach (static::$beforeSendingCallbacks as $callback) {
+            $callback($blueprint, $newRecipients);
```

Comment:
Should we offer a way for recipients to be modified other than via pass-by-reference? Maybe have the callbacks return a new list of recipients, since that would be the main purpose of inserting logic here?

Reply:
I would be in favor since that is indeed the main purpose here, also I think it's always better to avoid pass by reference where possible.